crates/bevy_animation/src/anim2d.rs

```diff
@@ -0,0 +1,51 @@
+//! 2d Animation for the game engine Bevy
+
+use bevy_math::{Quat, Vec2, Vec3};
+
+crate::anim_impl::anim_impl! {
+  /// 2d
+
+  type Rotation = f32;
+  type Translation = Vec2;
+  type Scale = Vec2;
+
+  struct AnimationClip2d;
+  struct AnimationPlayer2d;
+  struct AnimationPlugin2d;
+  struct Keyframes2d;
+  struct VariableCurve2d;
+}
+
+#[inline]
+fn rotation_to_quatlike(rotation: Rotation) -> Vec2 {
+    (rotation * 0.5).sin_cos().into()
+}
+#[inline]
+fn quatlike_to_quat(rotation: Vec2) -> Quat {
+    Quat::from_xyzw(0.0, 0.0, rotation.x, rotation.y)
+}
+#[inline]
+fn rotation_to_quat(rotation: Rotation) -> Quat {
+    Quat::from_rotation_z(rotation)
+}
+#[inline]
+fn translation_to_vec3(translation: Translation) -> Vec3 {
+    (translation, 0.0).into()
+}
+#[inline]
+fn scale_to_vec3(scale: Scale) -> Vec3 {
+    (scale, 1.0).into()
+}
+#[inline]
+fn slerp_quatlike(a: Vec2, b: Vec2, s: f32) -> Vec2 {
```

Comment:
This method is not slerp, it linearly interpolates the `cos` and `sin` of the angle, which is a different interpolation formula from slerp. Slerp linearly interpolates the angle then applies the `sin` over that, see https://en.wikipedia.org/wiki/Slerp. See also `slerp()` in `glam`: https://github.com/bitshifter/glam-rs/blob/e2e0d5179dafa057ff3dfb7b5bbeccd45d1b3c9b/src/f32/scalar/quat.rs#L566-L597